zcash_keys/src/keys.rs

```diff
@@ -71,6 +70,33 @@ pub mod sapling {
     }
 }
 
+/// A hash of a seed used for an HD account.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct HDSeedFingerprint([u8; BLAKE2B_LENGTH]);
+
+impl HDSeedFingerprint {
+    /// Generates the fingerprint from a given seed.
+    pub fn from_seed(seed: &SecretVec<u8>) -> Self {
+        const PERSONALIZATION: &[u8] = b"Zcash_HD_Seed_FP";
+        let hash = blake2bParams::new()
+            .personal(PERSONALIZATION)
+            .to_state()
+            .update(seed.expose_secret())
+            .finalize();
+        Self(hash.as_array().to_owned())
```

Comment:
This does not match [the Seed Fingerprints section of ZIP 32](https://zips.z.cash/zip-0032#seed-fingerprints), which I presume it intends to implement because it uses the personalization from that section. (It also doesn't match the pre-2020 specification, because while that also lacked a length prefix, the spec has always defined the fingerprint hash as `BLAKE2b-256`, not `BLAKE2b`).
```suggestion
        const PERSONALIZATION: &[u8] = b"Zcash_HD_Seed_FP";
        let hash = blake2bParams::new()
            .hash_length(32)
            .personal(PERSONALIZATION)
            .to_state()
            .update(&[u8::try_from(seed.len()).expect("ZIP 32 seeds MUST be at most 252 bytes")])
            .update(seed.expose_secret())
            .finalize();
        Self(hash.as_bytes().try_into().expect("length is correct above"))
```

Reply:
Nice catch!